File: frontend/lib/homeowner/types/provider.ts
/** Service-provider domain types — aligned with service_providers + project_contractor_matches. */

import type { ServiceCategorySlug } from "./project";

export interface ServiceProvider {
  id: string;
  businessName: string;
  avatarUrl: string | null;
  categories: ServiceCategorySlug[];
  location: string;
  rating: number; // 1–5
  reviewsCount: number;
  verified: boolean;
  /** 0–100 platform trust score. */
  trustScore: number;
  /** Median response time, human readable. */
  responseTime: string;
}

/** A provider recommended for a specific homeowner/project by the matching engine. */
export interface RecommendedProvider extends ServiceProvider {
  /** 0–100 compatibility from project_contractor_matches.compatibility_score. */
  matchScore: number;
  matchReason: string;
}

File: backend/app/homeowner/schemas/provider.py
"""Service-provider schemas — mirror frontend/lib/types/provider.ts."""

from app.shared.schemas.user import CamelModel


class ProviderOut(CamelModel):
    id: str
    business_name: str
    avatar_url: str | None = None
    categories: list[str]
    location: str
    rating: float
    reviews_count: int
    verified: bool
    trust_score: int
    response_time: str


class RecommendedProviderOut(ProviderOut):
    match_score: int
    match_reason: str
